### PR TITLE
Add global helper functions, update dependencies & readme, add DatoCMS

### PR DESCRIPTION
**Pull request by mwickett**
This moves some global helper functions up from the [hashicorp-www-middleman](https://www.github.com/hashicorp/hashicorp-www-middleman) project into this project for easier reuse across different projects.

### CODE REVIEW

**Comment by mwickett** [comment]
I've tested this with a modified (changes to `config.rb`) version of https://github.com/hashicorp/hashicorp-www-middleman and it's working well.

**Comment by mwickett** [comment]
@jescalan Tagging you here, I can't add you as a reviewer.

File: lib/middleman-hashicorp/extension.rb [hunk 1]
```diff
@@ -97,10 +111,104 @@ def initialize(app, options_hash = {}, &block)
...
+    # Middleware for rendering preact components
+    @app.use ReshapeMiddleware, component_file: options.reshape_component_file
+
+    # Activate middleman-dato if an api key is provided
+    # if (options.datocms_api_key)
+    #   require "middleman-dato"
+    #   @app.activate :dato,
+    #     token: options.datocms_api_key,
+    #     preview: ENV['ENV'] != 'production'
+    # end
```
Comment (jescalan) [inline lib/middleman-hashicorp/extension.rb]: Why commented out here?

File: lib/middleman-hashicorp/extension.rb [hunk 2]
```diff
@@ -97,10 +111,104 @@ def initialize(app, options_hash = {}, &block)
...
+
+    # compile js with webpack, css with postcss
+    @app.activate :external_pipeline,
+      name: 'assets',
+      command: "cd assets && NODE_ENV=#{ENV['ENV'] || 'development'}  ./node_modules/.bin/spike #{@app.build? ? :compile : :watch}",
+      source: 'assets/public'
```
Comment (jescalan) [inline lib/middleman-hashicorp/extension.rb]: Maybe we can make the assets path configurable but defaulted, since it would be easy?

**Review by jescalan — APPROVED** [review]
Looking great!

File: lib/middleman-hashicorp/extension.rb (hunk 1 above)
Comment (mwickett) [reply lib/middleman-hashicorp/extension.rb]: Meant to remove that. I was testing activating Dato in different places. It didn't work here.

File: lib/middleman-hashicorp/extension.rb (hunk 2 above)
Comment (mwickett) [reply lib/middleman-hashicorp/extension.rb]: Definitely - good call. Same as the Reshape component file.

File: lib/middleman-hashicorp/extension.rb (hunk 1 above)
Comment (mwickett) [reply lib/middleman-hashicorp/extension.rb]: Working on my Ruby 💪 haha

File: lib/middleman-hashicorp/extension.rb [hunk 3]
```diff
@@ -100,7 +117,113 @@ def initialize(app, options_hash = {}, &block)
     end
   end
 
+  expose_to_template :segmentId
+
+  # If a production Segment ID is provided, use it, otherwise use staging by default
+  def segmentId()
+    options.segment_production_key
+  end
```
Comment (jescalan) [inline lib/middleman-hashicorp/extension.rb]: Ah so we ended up moving the segment key into the JS config post-GDPR, so I think we can take this out now. Sorry for the backtrack!

**Review by jescalan — APPROVED** [review]
This looks perfect - good for merge after we drop the segmentId piece!

**Comment by mwickett** [comment]
Waiting on response back from Dato folks on slowdown in Dato 0.6.2 - see @schonfeld's [comment](https://github.com/hashicorp/hashicorp-www-middleman/pull/881#issuecomment-396579420)

**Comment by mwickett** [comment]
@jescalan Think this is ready to roll.

**Review by jescalan — APPROVED** [review]

**Pull request merged**